Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -0,0 +1,173 @@
+
+        for triage_item, alert_or_incident in data.values():
+            # validate we actually have an alert/incident before proceeding
+            if not alert_or_incident:
+                raise Exception(f'Triage item had no matching alert/incident data: {triage_item}')
+            alert_or_incident['status'] = triage_item['state']
+            alert_or_incident['triage_id'] = triage_item['id']
+            alert_or_incident['triage_raised_time'] = triage_item['raised']
+            alert_or_incident['triage_updated_time'] = triage_item['updated']
+            
+            #creating a custom json data to post into azure
+            azure_obj = {
+                **alert_or_incident,
+                'status': triage_item['state'],
+                'triage_id': triage_item['id'],
+                'triage_raised_time': triage_item['raised'],
+                'triage_updated_time': triage_item['updated'],
+                'comments': []
+            }
+
+            comment_data = self.DS_obj.get_triage_comments(triage_item['id'])
+            
+            for comment in comment_data:
+                if not comment['content']:
+                    continue
+                azure_obj['comments'].append({
+                    'user_name': comment['user']['name'],
+                    'content': comment['content'],
+                    'id': comment['id'],
+                    'created': comment['created']
+                })
+
+
+            self.AS_obj.post_data(json.dumps(azure_obj), constant.LOG_NAME)
+
+    def get_data(self):
+        """
+            getting the incident and alert data from digital shadows
+        """
+        triage_id = []
+        max_event_num = -1
+        item_data = []
+        event_data = []
+
+        try:
+            if(isinstance(self.event, int)):
+                event_data = self.DS_obj.get_triage_events_by_num(self.event)
+                #calculating the max event number from current batch to  use in next call
+                if event_data:
+                    max_event_num = max([e['event-num'] for e in event_data])
+
+            else:
+                event_data = self.DS_obj.get_triage_events(self.before_time, self.after_time)
+                #calculating the max event number from current batch to  use in next call
+                if event_data:
+                    max_event_num = max([e['event-num'] for e in event_data])
+                    logger.info("First poll from event number " + str(event_data[0]['event-num']))
+                    logger.info("Total number of events are " + str(len(event_data)))
+            
+            for event in event_data:
+                if(event is not None and event['triage-item-id'] not in triage_id):
+                    triage_id.append(event['triage-item-id'])
+
+            logger.info(triage_id)
+        except Exception:            
+            logger.exception("Error while getting triage data: ")
+        
+        if event_data:
```

Comment:
shouldn't this be `if triage_id:` ?

We aren't using `event_data` in the body of the if-block.

Reply:
It is checking if we have any events, if there are no events then no need to get triage items. I can use triage_id instead as well I guess since it seems more intuitive. Will change it.